src/main/java/com/atlassian/db/replica/internal/ForwardCall.java

```diff
@@ -1,18 +1,16 @@
 package com.atlassian.db.replica.internal;
 
-import com.atlassian.db.replica.api.*;
-import com.atlassian.db.replica.spi.*;
+import com.atlassian.db.replica.api.SqlCall;
+import com.atlassian.db.replica.api.context.RouteDecision;
+import com.atlassian.db.replica.spi.DatabaseCall;
 
-import java.sql.*;
+import java.sql.SQLException;
 
-public class ForwardCall implements DualCall {
-    @Override
-    public <T> T callReplica(SqlCall<T> call) throws SQLException {
-        return call.call();
-    }
+public class ForwardCall implements DatabaseCall {
 
     @Override
-    public <T> T callMain(SqlCall<T> call) throws SQLException {
+    public <T> T call(final SqlCall<T> call, RouteDecision context) throws SQLException {
```

Comment:
```suggestion
    public <T> T call(final SqlCall<T> call, RouteDecision decision) throws SQLException {
```

Reply:
It's outdated. I'm resolving the suggestion.